### PR TITLE
Address Sanitizer fixes (main)

### PR DESCRIPTION
**Pull request by korydraughn**
All unit tests and core tests passed.

All leaks fixed by this PR were discovered by running the unit tests with address sanitizer enabled.

We need to do the same thing with the core tests at some point.

### CODE REVIEW

File: server/api/src/rsRmColl.cpp [hunk 1]
```diff
@@ -746,14 +745,13 @@ rsMvCollToTrash( rsComm_t *rsComm, collInp_t *rmCollInp ) {
         }
     }
 
-    memset( &dataObjRenameInp, 0, sizeof( dataObjRenameInp ) );
+    dataObjCopyInp_t dataObjRenameInp{};
+    irods::at_scope_exit clear_dataObjCopyInp{[&dataObjRenameInp] { clearDataObjCopyInp(&dataObjRenameInp); }};
 
-    dataObjRenameInp.srcDataObjInp.oprType =
-        dataObjRenameInp.destDataObjInp.oprType = RENAME_COLL;
+    dataObjRenameInp.srcDataObjInp.oprType = dataObjRenameInp.destDataObjInp.oprType = RENAME_COLL;
```
Comment (trel) [inline server/api/src/rsRmColl.cpp:751]: on two lines instead of one?

optimize readability...

Comment (korydraughn) [reply server/api/src/rsRmColl.cpp]: Done

**Comment by korydraughn** [comment]
Addressed a few more leaks in the icommands.

Rerunning unit tests and core tests.

**Comment by korydraughn** [comment]
All unit tests passed.

Running core tests.

**Comment by korydraughn** [comment]
All core tests passed.

File: server/core/src/irods_file_object.cpp [hunk 2]
```diff
@@ -156,7 +156,6 @@ namespace irods {
         repl_requested_ = _rhs.repl_requested_;
         replicas_       = _rhs.replicas_;
         in_pdmo_        = _rhs.in_pdmo_;
-        replKeyVal( &_rhs.cond_input_, &cond_input_ );
```
Comment (alanking) [inline server/core/src/irods_file_object.cpp:159]: Is this being handled by the base class `operator=`? If not, are there any known effects of not copying the `cond_input_` here?

**Review by alanking — COMMENTED** [review]
Everything checks out. Had a question about one edit

File: server/core/src/irods_file_object.cpp (hunk 2 above)
Comment (korydraughn) [reply server/core/src/irods_file_object.cpp:159]: It's handled by the base class.

That line was causing the work to be done twice, which resulted in a leak.

Comment (alanking) [reply server/core/src/irods_file_object.cpp:159]: Great, thanks

**Comment by korydraughn** [comment]
Added 6634 as the issue number.

Any last comments before I add the pounds?

**Review by alanking — COMMENTED** [review]
Any reason these can't be squashed to fewer commits? I see three instances where there are identical commit messages one after the other. I don't feel strongly about this, just wondering

**Comment by korydraughn** [comment]
They can be squashed. I kept them separate to highlight individual fixes.

**Review by alanking — APPROVED** [review]
I leave it up to you or anybody else who wants to throw in an opinion, but I think this is fine. Add #s when ready, I say

**Comment by korydraughn** [comment]
Ok. I'll squash them. The changes are tiny and keeping them separate doesn't benefit anyone.

**Comment by trel** [comment]
Just easier to revert if kept separate.

**Pull request merged**